R/prep_data_forest_plots.R

```diff
@@ -4,24 +4,37 @@
 ##' names. It is generally not intended to be called directly but is used by
 ##' \code{\link{load_epidata}} when the data is loaded.
 ##'
-##' @param x data.frame containing a column called "parameter_type",  This will 
+##' @param x data.frame containing a column called "parameter_type",  This will
 ##' typically be the `params`data.frame from the output of \code{load_epireview}.
-##' @param parameter_type_full optional. User can specify the full value of a 
+##' @param parameter_type_full optional. User can specify the full value of a
 ##' parameter type not already included in the function.
 ##' @param parameter_type_short optional. Shorter value of parameter_type_full
 ##' @return data.frame with a new column called "parameter_type_short"
 ##' @export
 ##' @author Sangeeta Bhatia
-short_parameter_type <- function(x, parameter_type_full, parameter_type_short) {
-
-  x$parameter_type_short <- case_when(
-    x$parameter_type == "Reproduction number (Basic R0)" ~ "Basic (R0)",
-    x$parameter_type == "Reproduction number (Effective, Re)" ~ "Effective (Re)",
-    x$parameter_type == "Human delay - time symptom to outcome" &
-      x$riskfactor_outcome == "Death" ~ "Time symptom to outcome (Death)",
-    x$parameter_type == "Human delay - time symptom to outcome" &
-      x$riskfactor_outcome == "Other" ~ "Time symptom to outcome (Other)"
-  )
+short_parameter_type <- function(x, pathogen, par_types, parameter_type_full, parameter_type_short) {
+
+  if(pathogen == 'marburg'){
+    x$other_delay <- case_when(
+      x$parameter_type == "Human delay - time symptom to outcome" &
+        x$riskfactor_outcome == "Death" ~ "Time symptom to outcome (Death)",
+      x$parameter_type == "Human delay - time symptom to outcome" &
+        x$riskfactor_outcome == "Other" ~ "Time symptom to outcome (Other)",
+      TRUE ~ NA
+    )
+
+    x$parameter_type <- ifelse(x$parameter_type == 'Human delay - time symptom to outcome',
+                               paste0('Human delay - ', tolower(x$other_delay)), x$parameter_type)
+  } else if(pathogen == 'lassa'){
+    x$other_delay <- ifelse(x$parameter_type == 'Human delay - other human delay (go to section)',
+                            paste0(tolower(str_replace(x$other_delay_start, 'Other: ', '')), ' to ',
+                                   tolower(str_replace(x$other_delay_end, 'Other: ', ''))), NA)
+
+    x$parameter_type <- ifelse(x$parameter_type == 'Human delay - other human delay (go to section)',
+                               paste0('Human delay - ', x$other_delay), x$parameter_type)
+  }
+
+  x <- mrerge(x, par_types, by = 'parameter_type')
```

Comment:
There is a typo here (mrerge instead of merge), so this will obviously not work.